crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -50,11 +53,23 @@ async fn create(
     let req = request.into_inner();
     let key = key.into_inner();
 
-    if key.ends_with(".") {
-        log::error!("configuration key {key} cannot end with a '.' character.");
+    let regex = Regex::new(KEY_NAME_REGEX).map_err(|err| {
+        unexpected_error!("could not parse regex due to: {}", err.to_string())
+    })?;
+
+    if !regex.is_match(&key) {
+        return Err(bad_argument!(
+            "The key name {} is invalid, it should obey the regex {}",
+            key,
+            KEY_NAME_REGEX
+        ));
+    }
+
+    if key.starts_with(".") || key.ends_with(".") {
```

Comment:
@knutties Can we try including this in the regex itself?

Reply:
Good catch.  The regex gets a bit complicated to read - but I guess we can add a comment to explain that.  Will do this.

Reply:
@mahatoankitkumar - addressed with a regex.  Please take a look.

Reply:
@knutties Just a query, are we okay with key's name starting or ending with `-` or `_`?
Keys such as `_`, `_test_`, `----`, `____---`, `_____` would be valid for the above regex.